tools/build_api.py

```diff
@@ -433,30 +433,25 @@ def merge_region_list(region_list, destination, notify, padding=b'\xFF'):
 
             # make same assumption as in region builder; first segment must fit.
             # this is only to get a neat ToolException anyway, since IntelHex.merge will
-            # throw intelhex.AddressOverlapError if there's overlapping
+            # throw IntelHex.AddressOverlapError if there's overlapping
             part_size = 0
             for es in part.segments():
-                # Add padding in between segments starting from end of first segment
-                if (len(part.segments()) > 1 and (merged.maxaddr() != None)):
-                    pad_size = es[0] - (merged.maxaddr() + 1)
-                    merged.puts(merged.maxaddr()+1, padding * pad_size)
                 part_size += es[1] - es[0]
                 merged.merge(part[es[0]:_end_addr_inclusive(es[1])])
```

Comment:
Since this loop is no longer any different from `merged.merge(part)`, can we just do that?